src/App.js

```diff
@@ -1,16 +1,46 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import './App.css';
 
 const App = () => {
+  const initialCopy = chatMessages.map((chatEntry) => {
+    return { ...chatEntry };
+  });
```

Comment:
👀 We don't need to start out with a copy of the chat data (incidentally, this makes an additional copy which is subsequently thrown away every time the component renders). As long as the state modifying logic properly makes a copy that it uses for any modifications, there won't be any risk to the original data.